### DISCUSSION TITLE
Asus Ryujin pump and micro fan support

### DISCUSSION THREAD

**Discussion by Karmalakas** [Rem0o/FanControl.Releases#2311] [Ideas]
Was searching, but couldn't find much about this.

I have a Ryujin AIO cooler and besides the CPU fans on the radiator it also has a micro fan right on the pump. I'm not sure how it works, but it seems both pump and this micro fan are controlled purely by USB (there's a USB cable coming from the pump mount). AIO_PUMP connector on MB always gives full speed, so it's not possible to control it if I understood it correctly. Tried manual check, but none of the detected fans changed pump speed in Fan Control. I can add curves and everything via Asus Armoury Crate software (and I believe BIOS of Asus RoG Strix MB), but can't do anything via fan connectors.
![image](https://github.com/Rem0o/FanControl.Releases/assets/3602819/4b50402f-8f96-4d24-a741-d6dfe8990a63)

Any chance this could be implemented somehow in the Fan Control or is it in some way protected by Asus and not accessible at all?

**Comment by cotvwow** [comment]
Same issue here, would love to modify it via Fan Control, its a great tool!

**Comment by Karmalakas** [comment]
Here's an update from my side..

Since I started this topic, I found out that Fan Control supports plugins and there's one for water coolers. Sadly, I didn't manage to successfully install any of the plugins (tried 3 or 4).

Also, it seems all (or at least for the past few years) Asus MBs with ALC4080 audio are faulty. Mine RoG Strix z790-f Gaming was no exception. Also this MB provided only full speed for AIO_PUMP connector.

So, because of the audio not working properly, I replaced this MB with MSI (BTW also with ALC4080) and not only audio works as expected, but also AIO_PUMP (and all other fan connectors) can be changed to be DC or PWM controlled. All I had to do, is to set pump connector to DC controlled and set full speed in BIOS and now I can control it via Fan Control as any other fan without any add-ons. I believe this micro fan speed is related to the pump speed.

Just to note, that I don't care about any lights or info on the cooler display, so without Asus software it's dark and I always prefered there were no LEDs inside the PC case

It's wild, that you can do more (depending on what your needs are) with Asus cooler on non-Asus MBs 🤷‍♂️
